core/lib/Cardano/Address/Script.hs

```diff
@@ -222,6 +227,121 @@ scriptHashFromBytes bytes
     | BS.length bytes /= credentialHashSize = Nothing
     | otherwise = Just $ ScriptHash bytes
 
+-- | Encode a 'ScriptHash' to bech32 'Text' or hex if key role is unknown.
+--  If one wants to include, valid in governance roles only, additional byte
+--  as specified in CIP-0129, the function needs to be called with withByte=true.
+--
+-- @since 4.0.0
+scriptHashToText :: ScriptHash -> KeyRole -> Bool -> Text
+scriptHashToText (ScriptHash scriptHash) cred withByte = case cred of
+    Representative ->
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.drep) $ appendByte scriptHash
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.drep_script) scriptHash
+    CommitteeCold ->
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_cold) $ appendByte scriptHash
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_cold_script) scriptHash
+    CommitteeHot ->
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_hot) $ appendByte scriptHash
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_hot_script) scriptHash
+    Unknown ->
+        T.decodeUtf8 $ encode EBase16 scriptHash
+    _ ->
+        T.decodeUtf8 $ encode (EBech32 CIP5.script) scriptHash
+  where
+-- | In accordance to CIP-0129 (https://github.com/cardano-foundation/CIPs/tree/master/CIP-0129)
```

Comment:
I would move this comment into the haddock comment of the function itself.

Reply:
done